<!--
marker_track.launch: ROS Launch file to track AR marker for robust kinect and baxter calibration.
Requirements: ar_track_alvar installed and kinect2_bridge up and running
Author: Nishanth Koganti
Date: 2016/6/15
-->

<launch>
    <!-- define args -->
    <arg name="marker_size" default="4.5" />
  	<arg name="max_track_error" default="0.2" />
  	<arg name="max_new_marker_error" default="0.08" />
  	<arg name="output_frame" default="/kinect2_link" />
  	<arg name="cam_info_topic" default="/kinect2/qhd/camera_info" />
  	<arg name="cam_image_topic" default="/kinect2/qhd/image_color" />
    <arg name="save_path" default="$(find kinect_baxter_calibration)/files/" />
    <arg name="parameters_file"  default="$(find kinect_baxter_calibration)/files/marker_track.yaml" />

    <!-- marker tracking node -->
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" />

    <!-- marker tracking using marker_track.py -->
    <node pkg="kinect_baxter_calibration" type="marker_track.py" name="marker_track" output="screen" >
        <param name="save_path" value="$(arg save_path)" />
        <param name="parameters_file" value="$(arg parameters_file)" />
    </node>

    <!-- play trajectory during tracking -->
    <node pkg="baxter_examples" type="joint_position_file_playback.py" name="play_trajectory" args="-f $(find kinect_baxter_calibration)/files/marker_trajectory" />
</launch>
